src/spoke/BalanceSheet.sol

```diff
@@ -169,7 +169,10 @@ contract BalanceSheet is Auth, Multicall, Recoverable, IBalanceSheet, IBalanceSh
         if (shareQueue.isPositive || shareQueue.delta == 0) {
             shareQueue.delta += shares;
             shareQueue.isPositive = true;
-        } else if (shareQueue.delta > shares) {
+            if (shareQueue.delta == 0) {
+                shareQueue.isPositive = false;
+            }
+        } else if (shareQueue.delta >= shares) {
             shareQueue.delta -= shares;
             shareQueue.isPositive = false;
         } else {
```

Comment:
NOTE: add this `>=` on purpose to force the `delta = 0` in this branch where isPositive is set to false